speedtestcli.pm

```diff
@@ -135,34 +135,20 @@ sub pingone ($){
 	while (<$outh>) {
         $self->do_debug("output: ".$_);
         syslog("debug", "[Speedtest] output: ".$_);
-	    if (/$measurement/i) {
-            #sample output:
-            #Ping: 2.826 ms
-            #Download: 898.13 Mbit/s
-            #Upload: 420.01 Mbit/s
-            
-            my ($value, $unit) = /([0-9\.]+) ([A-Za-z\/]+)/;
-            #we're not always measuring seconds, but ProbeUnit() should provide the correct unit for the Y Axis
-            
-            #normalize the units to be in the same base.
-            my $factor = 1; 
-            $factor = 0.001 if($unit eq 'ms');
-            $factor = 1_000 if($unit eq 'Kbit/s' || $unit eq 'kbit/s');
-            $factor = 1_000_000 if($unit eq 'Mbit/s' || $unit eq 'mbit/s');
-            $factor = 1_000_000_000 if($unit eq 'Gbit/s' || $unit eq 'gbit/s');
-            
-            my $normalizedvalue = $value * $factor;
-            $self->do_debug("Got value: $value, unit: $unit -> $normalizedvalue\n");
-            syslog("debug","[Speedtest] Got value: $value, unit: $unit -> $normalizedvalue\n");
-            
-            push @times, $normalizedvalue;
-            last;
-	    }
-	}
-	waitpid $pid,0;
-	close $errh;
-	close $inh;
-	close $outh;
+        my ($value) = /"$measurement":{"bandwidth":([0-9]*)/;
+        my $normalizedvalue = $value * 8;
```

Comment:
Why are you multiplying by 8? Are you converting bytes to bits?

Reply:
Indeed. This way it aligns with the output from the prior probe. I checked this against the values on their results page.
The [following image](https://cloud.florianjensen.com/z8uxBjgY) is with me changing the probe from speedtest to "speedtestcli" on May 13, where data continues to align correctly while also matching the result on Speedtests.net for the same tests.